lib/ansible/modules/storage/dellemc/dellemc_pmax_createsg.py

```diff
@@ -0,0 +1,200 @@
+    required: false
+  srp_id:
+    description:
+      - "Storage Resource Pool Name, Default is set to SRP_1, if your system
+      has mainframe or multiple pools you can set this to a different value to
+      match your environment"
+    required: false
+  unispherehost:
+    description:
+      - "Fully Qualified Domain Name or IP address of Unisphere for PowerMax
+      host."
+    required: true
+  universion:
+    description:
+      - "Integer, version of unipshere software  e.g. 90"
+    required: true
+  verifycert:
+    description:
+      - "Boolean, security check on ssl certificates"
+    type: bool
+    required: true
+  vol_size:
+    description:
+      - "Integer value for the size of volumes.  All volumes will be created
+      with same size.  Use dellemc_pmax_addvol to add additional volumes if you
+      require different sized volumes once storage group is created."
+    required: true
+  volumeIdentifier:
+    description:
+      - "String up to 64 Characters no special character other than
+      underscore sets a label to make volumes easily identified on hosts can
+      run Dell EMC inq utility command to see this label is  inq -identifier
+      device_name"
+    required: false
+  user:
+    description:
+      - "Unisphere username"
+  password:
+    description:
+      - "password for Unisphere user"
+  workload:
+    description:
+      - "Block workload type, optional and can only be set on VMAX3 Hybrid
+      Storage Arrays. Default None."
+    required: false
+requirements:
+  - Ansible
+  - "Unisphere for PowerMax version 9.0 or higher."
+  - "VMAX All Flash, VMAX3, or PowerMax storage Array."
+  - "PyU4V version 3.0.0.8 or higher using PIP python -m pip install PyU4V"
+'''
+EXAMPLES = '''
+---
+- name: "Provision Storage For DB Cluster"
+  connection: local
+  hosts: localhost
+  vars:
+    array_id: 000197600156
+    password: smc
+    sgname: Ansible_SG
+    unispherehost: "192.168.1.123"
+    universion: "90"
+    user: smc
+    verifycert: false
+
+  tasks:
+    - name: "Create New Storage Group and add data volumes"
+      dellemc_pmax_createsg:
+        array_id: "{{array_id}}"
+        cap_unit: GB
+        num_vols: 1
+        password: "{{password}}"
+        sgname: "{{sgname}}"
+        slo: Diamond
+        srp_id: SRP_1
+        unispherehost: "{{unispherehost}}"
+        universion: "{{universion}}"
+        user: "{{user}}"
+        verifycert: "{{verifycert}}"
+        vol_size: 1
+        workload: None
+        volumeIdentifier: 'REDO'
+'''
+RETURN = '''
+'''
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    changed = False
+    module = AnsibleModule(
+        argument_spec=dict(
+            sgname=dict(type='str', required=True),
+            unispherehost=dict(required=True),
+            universion=dict(type='int', required=False),
+            verifycert=dict(type='bool', required=True),
+            user=dict(type='str', required=True),
```

Comment:
If user, password, parameters are common to all other modules then it is better to create common Argument spec like [this](https://github.com/ansible/ansible/blob/013c42b14fa06ff6355d8a9d52430f4cb1262047/lib/ansible/module_utils/vmware.py#L448) one

Reply:
I'll rework these changes and push once retested.